client/src/retry.rs

```diff
@@ -184,10 +201,12 @@ where
             match self.backoff.next_backoff() {
```

Comment:
You should switch on the code before consuming the next backoff.
Something like:

```
let backoff = if e.code() == Code::ResourceExhausted {
    self.throttle_backoff.next_backoff()
} else {
    self.backoff.next_backoff()
}
match backoff {
   // cases for None / Some
}
```

Reply:
If you want to match TS, you could consume both backoffs and then do the switch.

Reply:
Hmm.. yeah, I see it matches Go now.

I like the TS approach where the attempt increments in both throttlers but it's really not critical IMHO.

Reply:
I tried to match the server behavior and the go sdk behavior

```
if next = r.NextBackOff(); next == done {
	return err
}

if err == ctx.Err() || !isRetryable(err) {
	return err
}

if _, ok := err.(*serviceerror.ResourceExhausted); ok {
	next = util.Max(next, t.NextBackOff())
}
```